backend/de.metas.business/src/main/java/de/metas/payment/paymentterm/impl/PaymentTermRepository.java

```diff
@@ -131,12 +155,68 @@ public Optional<PaymentTermId> retrievePaymentTermId(@NonNull final PaymentTermQ
 	@Override
 	public boolean isAllowOverrideDueDate(@NonNull final PaymentTermId paymentTermId)
 	{
-		return Services.get(IQueryBL.class)
+		final PaymentTerm paymentTerm = getById(paymentTermId);
+		return  paymentTerm.isAllowOverrideDueDate();
+	}
+
+
+	private PaymentTermMap getIndexedPaymentTerms()
+	{
+		return cache.getOrLoad(0, this::retrieveIndexedPaymentTerms);
+	}
+
+	private PaymentTermMap retrieveIndexedPaymentTerms()
+	{
+		final ImmutableList<PaymentTerm> paymentTerms = queryBL
 				.createQueryBuilder(I_C_PaymentTerm.class)
-				.addEqualsFilter(I_C_PaymentTerm.COLUMNNAME_C_PaymentTerm_ID, paymentTermId)
-				.addEqualsFilter(I_C_PaymentTerm.COLUMNNAME_IsAllowOverrideDueDate, true)
+				.addOnlyActiveRecordsFilter()
+				.orderBy(I_C_PaymentTerm.COLUMNNAME_IsDefault)
```

Comment:
why this? 
i mean,why have `.orderBy(I_C_PaymentTerm.COLUMNNAME_IsDefault)` ? it puts the ones with default=false first..which is that needed?